### ISSUE TITLE
🐛 [Bug] GDT Server incompatible with Communities

### ISSUE THREAD

**Issue by ulfgebhardt** [gradido/gradido#866]
## 🐛 Bugreport

I have thought about our GDT Server in relation to our Community-Approach:
- Communities need the GDT Server to show the GDT Balance & Transactions of a User
- The GDT Server is only reachable internally and therefore not available for external Communities
- The GDT Server cannot be public since you can simply query the balance of a User with its Email

This means if we expose the GDT Server to the public and therefore to other Communities an Attacker can query if an EMail has donated to Gradido and how much. I consider this a major breach of privacy and therefore we need to find a better Solution.

Any Ideas?

**Comment by einhornimmond** [comment]
In my original plan for GDT transaction users gets an extra address (public key) and we make real blockchain transactions for gdt to that address. But for this to work we need Multiple Addresses which isn't finished, and we must finally decide if we want use blockchain at all or not. 
When we not use Blockchain, I don't know how we would handle that.

**Comment by BerndHueckstaedt** [comment]
I thought, we use Blockchain / IOTA for permanent storage and / or backup?

**Comment by einhornimmond** [comment]
Ja, dachte ich auch, aber moritz bat mich drüber nachzudenken das vielleicht doch lieber wegzulassen um die Komplexität zu verringern. Und später die Sicherheit wieder über andere Ansätze herzustellen. 
Ich habe drüber nachgedacht und ich bin der Meinung das es kein Sinn macht auf blockchain zu verzichten.
Wenn überhaupt wird es vermutlich sehr aufwändig dieselben Sicherheitsfunktionen von einem Security Layer zu erhalten der irgendwie drauf gesetzt wird. 

So yes we will use blockchain, sorry @Mogge

**Comment by einhornimmond** [comment]
We need #98 for that.

**Comment by ulfgebhardt** [comment]
This Issue is not about the use of Blockchain, this issue is about GDT in its current state. Even if we use a Blockchain for this and start developing on this, which will set us back quite a bit regarding time schedule, it does not solve the problem of anonymity. If we use an anonymous ID for a user, how does the other community understand which GDT-Account is has to get from the Blockchain.

Do we want to start developing a Blockchain for GDT?
Would it solve the actual problem?
Is it worth the immense time effort to do so?
Is there a simpler solution?

**Comment by einhornimmond** [comment]
The GDT Transactions are similar to the GDD transactions and go on the blockchain where the user is currently. 
The plan was to use derived addresses. So the user has a main key-pair from which additional addresses can be derived. Without knowing the private key, only the public key can derived for named sub-accounts for example with the name GDT. 
With knowing the private key of main key-pair the user can derive the private key for this subaccount to transform the GDT to GDD.
So if you know the main public key from the user you can derive the gdt address for this user. 
But not the other way round. 

So we don't need starting developing a blockchain for gdt. 
We need the multiple addresses. And then we need a special transaction type with Bernd can transfer the GDT to the user.
And later a special transaction to transform GDT to GDD. For this we need rules to decide and implement so that GDT cannot unbalance the GDDs.

**Comment by einhornimmond** [comment]
A transitional solution for the anonymous problem could be using the elopage user ids instead of email addresses to get the GDT for a user.

**Comment by ulfgebhardt** [comment]
![gdt_server_api](https://user-images.githubusercontent.com/1238238/134724882-59492b2d-dc45-48b6-be5d-1ea5f083af2c.png)

**Comment by clauspeterhuebner** [comment]
@ulfgebhardt 
this ticket will be part of the GradidoID-Epic and needs changes in the GDT-server and api requested from apollo.
On my opinion we should use the users UUID, but as soon as we support several communities there must be a technical identifier for the community too. The community url will be unique, but it could be changed.
So i suggest to introduce a community-UUID and in conjunction with the users-UUID we will be prepared for all kinds in the future to address a GDT-user on a technical way.
